tests/unit/monitor/data_monitoring/alerts/test_data_monitoring_alerts.py

```diff
@@ -0,0 +1,131 @@
+import json
+
+import pytest
+
+from elementary.monitor.alerts.model_alert import ModelAlertModel
+from elementary.monitor.alerts.source_freshness_alert import SourceFreshnessAlertModel
+from elementary.monitor.alerts.test_alert import TestAlertModel
+from elementary.monitor.api.alerts.schema import AlertsSchema
+from elementary.monitor.data_monitoring.alerts.integrations.slack.slack import (
+    SlackIntegration,
+)
+from tests.mocks.api.alerts_api_mock import MockAlertsAPI
+from tests.mocks.data_monitoring.data_monitoring_alerts_mock import (
+    DataMonitoringAlertsMock,
+)
+
+
+def test_get_integration_client(data_monitoring_alerts_mock: DataMonitoringAlertsMock):
+    integration = data_monitoring_alerts_mock._get_integration_client()
+    assert isinstance(integration, SlackIntegration)
+
+
+def test_fetch_data(data_monitoring_alerts_mock: DataMonitoringAlertsMock):
+    mock_alerts_api = MockAlertsAPI()
+    alerts_data = data_monitoring_alerts_mock._fetch_data(days_back=1)
+    assert isinstance(alerts_data, AlertsSchema)
+    assert alerts_data.json(sort_keys=True) == mock_alerts_api.get_new_alerts(
+        days_back=1
+    ).json(sort_keys=True)
+
+
+def test_format_alerts(data_monitoring_alerts_mock: DataMonitoringAlertsMock):
+    alerts = data_monitoring_alerts_mock._fetch_data(days_back=1)
+    formatted_alerts = data_monitoring_alerts_mock._format_alerts(alerts)
+    test_alerts = [
+        alert for alert in formatted_alerts if isinstance(alert, TestAlertModel)
+    ]
+    model_alerts = [
+        alert for alert in formatted_alerts if isinstance(alert, ModelAlertModel)
+    ]
+    source_freshness_alerts = [
+        alert
+        for alert in formatted_alerts
+        if isinstance(alert, SourceFreshnessAlertModel)
+    ]
+    assert json.dumps(
+        [alert.id for alert in test_alerts], sort_keys=True
+    ) == json.dumps(["alert_id_2", "alert_id_3", "alert_id_4"])
+    assert json.dumps(
+        [alert.id for alert in model_alerts], sort_keys=True
+    ) == json.dumps(["alert_id_2", "alert_id_3", "alert_id_4"])
+    assert json.dumps(
+        [alert.id for alert in source_freshness_alerts], sort_keys=True
+    ) == json.dumps(["alert_id_2", "alert_id_3", "alert_id_4"])
```

Comment:
maybe we want to check more fields are formatted properly? could be nice to add a verification that nothing is omitted in the formatting

Reply:
So I don't check here that the fields got formatted properly, but that the right alerts got formatted to the relevant format.
Currently the Formatting is straight forward so I didn't add a test for that explicit